Packages/Core/Editor/Generator/Generator.cs

```diff
@@ -1,164 +1,232 @@
+                    solvers = new SortedList<string, Solver>(readOnlySolvers);
                 }
+            }
+            _solversPopulated = true;
 
-                var resolvedRelativeFilePath = Templating.ResolveVariables(templateVariables: templateVariables, toResolve: relativeFilePath);
-                var template = File.ReadAllText(templatePath);
-                var filePath = Path.Combine(baseWritePath, resolvedRelativeFilePath);
-                var fileDirectory = Path.GetDirectoryName(filePath);
+            _keys = new List<string>(solvers.Keys);
 
-                // Create write directory
-                Directory.CreateDirectory(fileDirectory);
+            _templates = new List<Template>(solvers.Values.Count);
+            _options = new List<bool>(solvers.Values.Count);
+            _scripts = new List<MonoScript>(solvers.Values.Count);
+            for (int i = 0; i < solvers.Values.Count; i++)
+            {
+                var solver = solvers.Values[i];
 
-                // Adjust content
-                var content = Templating.ResolveVariables(templateVariables: templateVariables, toResolve: template);
-                content = Templating.ResolveConditionals(template: content, trueConditions: templateConditions);
-                content = RemoveDuplicateNamespaces(template: content);
+                _templates.Add(solver.template);
+                _options.Add(solver.option);
+                _scripts.Add(solver.script);
+            }
+        }
 
-                // Write to file
-                File.WriteAllText(filePath, content);
-                AssetDatabase.ImportAsset(filePath);
+        public void OnAfterDeserialize()
+        {
+            generatedType = Type.GetType(_assemblyQualifiedName ?? string.Empty);
+
+            solvers.Clear();
+            for (int i = 0; i < _keys.Count; i++)
+            {
+                var key = _keys[i];
+                var solver = new Solver(_templates[i])
+                {
+                    option = _options[i],
+                    script = _scripts[i],
+                };
+
+                solvers.Add(key, solver);
             }
+
+            _solversPopulated = false;
         }
+        #endregion
 
+        #region Generation
         /// <summary>
-        /// Removes duplicate namespaces, given content from a template.
+        /// Generate scripts inside the generators directory from the generators templates.
         /// </summary>
-        /// <param name="template">The content template to remove namespace from.</param>
-        /// <returns>A copy of `content`, but without duplicate namespaces.</returns>
-        private static string RemoveDuplicateNamespaces(string template)
+        public void Generate()
         {
-            var currentIndex = 0;
-            var countNamespaces = new Dictionary<string, int>();
-            while (true)
+            var assetPaths = new string[solvers.Values.Count];
```

Comment:
If we generate a set of Atoms of one type, and then change the type and regenerate the old Atoms are still there, but unreferenced by the the generator and the generator creates a new set of Atoms. In my mind we should delete the old Atoms of the old type and then create the new ones. Maybe we should show a prompt before regenerating as well, where one could choose to delete or to keep the Atoms (unreferenced by any generator)?

Reply:
Unfortunately, since the MonoScript references will have to be removed because they are broken, I can't implement this feature.